mindsdb/integrations/handlers/anomaly_detection_handler/anomaly_detection_handler.py

```diff
@@ -0,0 +1,172 @@
+    df[categorical_columns] = df[categorical_columns].apply(lambda x: x.cat.codes)
+    df = pd.get_dummies(df, columns=categorical_columns)
+    numeric_columns = list(df.select_dtypes(include=["number"]).columns.values)
+    df[numeric_columns] = (df[numeric_columns] - df[numeric_columns].mean()) / df[numeric_columns].std()
+    return df
+
+
+def get_model_names(using_args):
+    """Get the model names from the using_args. Model names is a list of model names to train.
+    If the model is not an ensemble, it only contains one model"""
+    model_names = anomaly_type_to_model_name(using_args["anomaly_type"]) if "anomaly_type" in using_args else None
+    model_names = using_args["model_name"] if "model_name" in using_args else model_names
+    model_names = using_args["ensemble_models"] if "ensemble_models" in using_args else model_names
+    model_names = [model_names] if model_names is None else model_names
+    model_names = [model_names] if type(model_names) is str else model_names
+    return model_names
+
+
+class AnomalyDetectionHandler(BaseMLEngine):
+    """Integration with the PyOD and CatBoost libraries for
+    anomaly detection. Both supervised and unsupervised.
+    """
+
+    name = "anomaly_detection"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.generative = True  # makes unsupervised learning work
+
+    def create(self, target, df, args={}):
+        """Train a model and save it to the model storage"""
+        using_args = args["using"]
+        model_type = using_args["type"] if "type" in using_args else None
+
+        model_names = get_model_names(using_args)
+
+        model_save_paths = []
+        model_targets = []
+        model_class_names = []
+        for model_name in model_names:
+            model = choose_model(df, model_name=model_name, model_type=model_type, target=target)
+            this_model_target = "outlier" if target is None else target  # output column name for unsupervised learning
+            save_path = "model.joblib" if model_name is None else model_name + ".joblib"
+            dump(model, save_path)
+            model_save_paths.append(save_path)
+            model_targets.append(this_model_target)
+            model_class_names.append(model.__class__.__name__)
+        model_args = {"model_path": model_save_paths, "target": model_targets, "model_name": model_class_names}
+        self.model_storage.json_set("model_args", model_args)
+
+    def predict(self, df, args={}):
+        """Load a model from the model storage and use it to make predictions"""
+        model_args = self.model_storage.json_get("model_args")
+        results_list = []
+        if "__mindsdb_row_id" in df.columns:
+            df = df.drop("__mindsdb_row_id", axis=1)
+        if model_args["target"][0] in df.columns:
+            df = df.drop(model_args["target"], axis=1)
+        for model_path in model_args["model_path"]:
+            model = load(model_path)
```

Comment:
The similar as for 'create' method:
- `base_path = self.model_storage.folder_get('context')` before circle
- `model = load(os.path.join(base_path, model_path))` instead this line